pkg/register/restResolver.go

```diff
@@ -168,7 +205,10 @@ func (c *RestResolverClient) RegisterDocument(ctx context.Context, document *Reg
 		}
 		return &ResolverError{err: err, errType: ConnectionError}
 	}
-	defer response.Body.Close()
+	defer func() {
+		response.Body.Close()
+		c.meter.RecordConnectionReleased()
+	}()
```

Comment:
minor: you can stack `defers` so this might look nicer as it was, and having `defer c.meter.RecordConnectionReleased()` after line 193